samples/subsys/ipc/rpmsg_multi_instance/remote/src/main.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/ipm.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <init.h>
+
+#include <ipc/rpmsg_multi_instance.h>
+
+#define IPM_WORK_QUEUE_STACK_SIZE 1024
+K_THREAD_STACK_DEFINE(ipm_stack_area_1, IPM_WORK_QUEUE_STACK_SIZE);
+K_THREAD_STACK_DEFINE(ipm_stack_area_2, IPM_WORK_QUEUE_STACK_SIZE);
+
+#define APP_TASK_STACK_SIZE (1024)
+K_THREAD_STACK_DEFINE(thread_stack_1, APP_TASK_STACK_SIZE);
+K_THREAD_STACK_DEFINE(thread_stack_2, APP_TASK_STACK_SIZE);
+static struct k_thread thread_data_1, thread_data_2;
+
+K_SEM_DEFINE(bound_ept1_sem, 0, 1);
+K_SEM_DEFINE(bound_ept2_sem, 0, 1);
+
+K_SEM_DEFINE(data_rx1_sem, 0, 1);
+K_SEM_DEFINE(data_rx2_sem, 0, 1);
+
+unsigned int received_data_1, received_data_2;
+
+static struct rpmsg_mi_ctx ctx_1, ctx_2;
+struct rpmsg_mi_ept ept_1, ept_2;
+
+
+int boud_cb(void *priv)
+{
+	struct rpmsg_mi_ept *ept = (struct rpmsg_mi_ept *)priv;
+
+	if (ept == &ept_1) {
+		k_sem_give(&bound_ept1_sem);
+	} else if (ept == &ept_2) {
+		k_sem_give(&bound_ept2_sem);
+	}
+	return 0;
+}
+
+int received_cb(void *data, size_t len, void *priv)
+{
+	struct rpmsg_mi_ept *ept = (struct rpmsg_mi_ept *)priv;
+
+	if (ept == &ept_1) {
+		received_data_1 = *((unsigned int *)data);
```

Comment:
It looks you assumed here that endianness for both ends of IPC communication is the same. I'm afraid that's not always true. Could be safer to decode from network to host endianness here and from host to network when sending integer value.

Reply:
In other IPC samples, this is done similarly. For consistency with other examples, this is how it was done. This can be changed if desired.

Reply:
I'm fine to keep it as it is. Just found a potential problem during review.

Reply:
Sure, when working with architectures of different endianness, it is important to keep this in mind.